src/o365/spfx/commands/project/project-rename.ts

```diff
@@ -0,0 +1,193 @@
+import commands from '../../commands';
+import Command, {
+  CommandOption, CommandAction, CommandError, CommandValidate
+} from '../../../../Command';
+import GlobalOptions from '../../../../GlobalOptions';
+import { BaseProjectCommand } from './base-project-command';
+import * as path from 'path';
+import * as fs from 'fs';
+const uuid = require('uuid');
+
+const vorpal: Vorpal = require('../../../../vorpal-init');
+
+interface CommandArgs {
+  options: Options;
+}
+
+interface Options extends GlobalOptions {
+  newName: string;
+  generateNewId?: boolean;
+}
+
+class SpfxProjectRenameCommand extends BaseProjectCommand {
+  public constructor() {
+    super();
+  }
+
+  public static ERROR_NO_PROJECT_ROOT_FOLDER: number = 1;
+
+  public get name(): string {
+    return commands.PROJECT_RENAME;
+  }
+
+  public get description(): string {
+    return 'Rename SharePoint Framework project';
+  }
+
+  public getTelemetryProperties(args: CommandArgs): any {
+    const telemetryProps: any = super.getTelemetryProperties(args);
+    telemetryProps.newName = (!(!args.options.newName)).toString();
```

Comment:
Since `newName` is required, there is no point in tracking its usage because it would always show up as specified